src/layout/layout.rs

```diff
@@ -1212,5 +1267,176 @@ mod tests {
                 ]
             );
         }
+
+        #[test]
+        fn fill_fixed() {
+            // fixed doesn't panic when results don't match exactly
+            let [a, b, c] = Rect::new(0, 0, 100, 1).split(&Layout::horizontal([
+                Fixed(33),
+                Fixed(33),
+                Fixed(33),
+            ]));
+            assert_eq!([a.width, b.width, c.width], [33, 33, 34]);
+
+            // cassowary implementation tends to put excess in last variable
+            let [a, b, c] = Rect::new(0, 0, 100, 1).split(&Layout::horizontal([
+                Fixed(25),
+                Fixed(25),
+                Fixed(25),
+            ]));
+            assert_eq!([a.width, b.width, c.width], [25, 25, 50]);
+
+            // fixed with min and max
+            let [a, b, c] =
+                Rect::new(0, 0, 100, 1).split(&Layout::horizontal([Min(25), Fixed(25), Max(25)]));
+            assert_eq!([a.width, b.width, c.width], [50, 25, 25]);
+
+            // fixed with percent and ratio
+            let [a, b, c] = Rect::new(0, 0, 100, 1).split(&Layout::horizontal([
+                Percentage(25),
+                Fixed(25),
+                Ratio(1, 4),
+            ]));
+            assert_eq!([a.width, b.width, c.width], [25, 25, 50]);
+
+            // 3 lengths for reference
+            // cassowary implementation tends to put excess in last variable
+            let [a, b, c] = Rect::new(0, 0, 100, 1).split(&Layout::horizontal([
+                Length(25),
+                Length(25),
+                Length(25),
+            ]));
+            assert_eq!([a.width, b.width, c.width], [25, 25, 50]);
+
+            // fixed with length
+            // this test gives different results when run over and over again
+            //
+            // assert_eq!([a.width, b.width], [25, 50])
+            // assert_eq!([a.width, b.width], [50, 25])
+            //
+            // So we check just that the last value is what we want it to be
+            let [_a, _b, c] = Rect::new(0, 0, 100, 1).split(&Layout::horizontal([
+                Length(25),
+                Length(25),
+                Fixed(25),
+            ]));
+            assert_eq!(c.width, 25);
+
+            let layout = Layout::default()
+                .constraints([Min(20), Fixed(1), Length(100)])
+                .direction(Direction::Horizontal)
+                .split(Rect::new(0, 0, 50, 1));
```

Comment:
add `.map(Rect::width)` to simplify the tests to a single assertion over an array of u16

Reply:
Done.